HW4_Bobkov/protein_tools.py

```diff
@@ -0,0 +1,220 @@
+        if aa not in res:
+            res[aa] = 1
+        else:
+            res[aa]+=1
+    res.update((key, round(value/l*100, 2)) for key, value in res.items())
+    res={key: value for key, value in sorted(res.items(), key=lambda item: item[1], reverse=True)}
+    return res
+
+
+def compare_pattern(sequence: str, pattern: str)->bool:
+    """
+    Compare a given pattern to a fragment of sequence of the same length
+    arguments:
+    - sequence (str): sequence fragment to compare with the pattern
+    - pattern (str): pattern for comparison
+    return:
+    - (bool): whether pattern and fragment match
+    """
+    for i in range(0,len(sequence)):
+        if not sequence[i]==pattern[i]:
+            return False
+            break
+    return True
+    
+def find_pattern(sequences: list, pattern: str)->dict:
+    """
+    Find all non-overlaping instances of a given pattern in sequences
+    arguments:
+    - sequences (list): sequences to find the pattern in
+    - pattern (str): pattern in question
+    return
+    - finds(dict): dictionary with sequences as keys and lists of indexes of patterns and the number of patterns as values
+    """
+    finds={}
+    for j in range(0, len(sequences)):
+        find=[]
+        for i in range(0, len(sequences[j])):
+            if compare_pattern(sequences[j][i:i+len(pattern)], pattern):
+                find.append(i)
+                i+=len(pattern)
+            else:
+                continue
+        finds[sequences[j]]=[len(find)]+find
+    return finds
+
+
+def transform_to_DNA_code(protein):
+    """
+    Transforming of an amino acid sequence/protein to DNA sequence
+    :param protein: amino acid sequence of protein
+    :return: sequence of protein in the DNA sequence form 
+    """
+    retrnaslation_dict = {
+        'F': 'TTC', 'f': 'ttc',
+        'L': 'TTA', 'l': 'tta',
+        'S': 'TCG', 's': 'tcg',
+        'Y': 'TAC', 'y': 'tac',
+        'C': 'TGC', 'c': 'tgc',
+        'W': 'TGG', 'w': 'tgg',
+        'P': 'CCC', 'p': 'ccc',
+        'H': 'CAT', 'h': 'cat',
+        'Q': 'GAA', 'q': 'gaa',
+        'R': 'CGA', 'r': 'cga',
+        'I': 'ATT', 'i': 'att',
+        'M': 'ATG', 'm': 'atg',
+        'T': 'ACC', 't': 'acc',
+        'N': 'AAT', 'n': 'aat',
+        'K': 'AAA', 'k': 'aaa',
+        'V': 'GTT', 'v': 'gtt',
+        'A': 'GCA', 'a': 'gca',
+        'D': 'GAT', 'd': 'gca',
+        'E': 'GAG', 'e': 'gag',
+        'G': 'GGG', 'g': 'ggg'
+    }
+
+    return ''.join([retrnaslation_dict[i] for i in protein])
+
+
+def rename_three_letter_name (seqs: list, sep = '')->list:
+    """
+    Transform into a three-letter amino acids entry.
+    arguments:
+        - seqs (list): list of sequences for transforming to three-letter entire
+        - sep (str): separator between aminoacids, default = ''
+    return:
+        - list: transformed sequences with separators
+    """
+    res=[]
+    threel = {'A': 'ALA', 'R': 'ARG', 'N': 'ASN', 'D': "ASP", 'V': 'VAL', 
```

Comment:
Опять же это дело лучше бы вынести в константы. 
```suggestion
    AMINOACIDS_CODES = {'A': 'ALA', 'R': 'ARG', 'N': 'ASN', 'D': "ASP", 'V': 'VAL', 
```